src/main/java/callhub/connect/data_access/FileDataAcessObject.java

```diff
@@ -0,0 +1,78 @@
+package callhub.connect.data_access;
+
+import callhub.connect.entities.FileDocument;
+import callhub.connect.entities.Session;
+import callhub.connect.use_case.file.FileDataAccessInterface;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class FileDataAcessObject implements FileDataAccessInterface {
+    @Autowired
+    private DocumentRepository documentRepository;
+
+    @Autowired
+    private SessionRepository sessionRepository;
+
+    /**
+     * Searches for and returns a FileDocument by its ID.
+     *
+     * @param id The ID of the FileDocument to be retrieved.
+     * @return An Optional containing the FileDocument if found, or an empty Optional if not found.
+     */
+    @Override
+    public Optional<FileDocument> findDocumentByID(String id) {
+        return documentRepository.findById(id);
+    }
+
+    /**
+     * Attempts to save a FileDocument to the document repository.
+     *
+     * This method tries to save the provided FileDocument object to the repository.
+     * If successful, it returns an Optional containing the saved FileDocument.
+     * If any exception occurs during the save process, it returns an empty Optional.
+     *
+     * @param file The FileDocument to be saved.
+     * @return An Optional containing the saved FileDocument, or an empty Optional in case of failure.
+     */
+    @Override
+    public Optional<FileDocument> uploadFile(FileDocument file) {
+        try {
+            return Optional.of(documentRepository.save(file));
+        } catch (Exception e) {
+            return Optional.empty();
```

Comment:
out of curiosity, why return an empty Optional here instead of an exception?

Reply:
its handled later i think an if checks for an empty